engine/state_stream/api_test.go

```diff
@@ -0,0 +1,120 @@
+package state_stream
+
+import (
+	"bytes"
+	"context"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/engine/common/rpc/convert"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/testutils"
+	"github.com/onflow/flow-go/module/executiondatasync/execution_data"
+	downloadermock "github.com/onflow/flow-go/module/executiondatasync/execution_data/mock"
+	protocol "github.com/onflow/flow-go/state/protocol/mock"
+	"github.com/onflow/flow-go/storage"
+	storagemock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type Suite struct {
+	suite.Suite
+
+	blocks     *storagemock.Blocks
+	headers    *storagemock.Headers
+	seals      *storagemock.Seals
+	results    *storagemock.ExecutionResults
+	downloader *downloadermock.Downloader
+}
+
+func TestHandler(t *testing.T) {
+	suite.Run(t, new(Suite))
+}
+
+func (suite *Suite) SetupTest() {
+	rand.Seed(time.Now().UnixNano())
+	header := unittest.BlockHeaderFixture()
+	params := new(protocol.Params)
+	params.On("Root").Return(header, nil)
+	suite.blocks = new(storagemock.Blocks)
+	suite.headers = new(storagemock.Headers)
+	suite.seals = new(storagemock.Seals)
+	suite.results = new(storagemock.ExecutionResults)
+
+	suite.downloader = new(downloadermock.Downloader)
```

Comment:
if you use the mockery `New*` constructors for these, it will automatically `AssertExpectations` when the test finishes